base/uk.ac.stfc.isis.ibex.ui/src/uk/ac/stfc/isis/ibex/ui/tables/DataboundTable.java

```diff
@@ -324,11 +332,11 @@ public Table table() {
 
     /**
      * Viewer should be created with tableComposite() as it's parent.
-     * 
+     * @param addEmptyRow to create table with one "permanent" empty row 
      * @return viewer
      */
-    protected TableViewer createViewer() {
-	return viewer = new TableViewer(tableComposite, tableStyle);
+    protected TableViewerEmptyRow createViewer(boolean addEmptyRow) {
+	return viewer = new TableViewerEmptyRow(tableComposite, tableStyle, addEmptyRow);
```

Comment:
lets use an if statement here and create a TableViewer if addEmptyRow is false and a TableViewerEmptyRow if it's true